src/components/map-view.jsx

```diff
@@ -0,0 +1,302 @@
+        let newTableEquipment = {
+            id: equipment.equipmentId,
+            type: equipment.equipmentType,
+            changed: !tableEquipment.changed,
+        };
+        onTableEquipementChanged(newTableEquipment);
+        onChangeTab(1); // switch to spreadsheet view
+    }
+
+    function getMapWitdh(displayMode) {
+        if (displayMode === STUDY_DISPLAY_MODE.DRAW) {
+            return '80%';
+        } else {
+            return '100%';
+        }
+    }
+    const onCancelFunction = useCallback(() => {
+        networkMapref.current.getMapDrawer().deleteAll();
+        dispatch(setStudyDisplayMode(STUDY_DISPLAY_MODE.MAP));
+    }, [dispatch]);
+    return (
+        <ReactFlowProvider>
+            <Box sx={styles.table}>
+                <Box sx={styles.horizontalToolbar}>
+                    <HorizontalToolbar />
+                </Box>
+                <div
+                    style={{
+                        display: 'flex',
+                        flexDirection: 'row',
+                        flexGrow: 1,
+                        overflow: 'hidden',
+                    }}
+                >
+                    <div
+                        style={{
+                            display:
+                                studyDisplayMode === STUDY_DISPLAY_MODE.MAP ||
+                                studyDisplayMode === STUDY_DISPLAY_MODE.DRAW
+                                    ? 'none'
+                                    : null,
+                            width: getMapWitdh(studyDisplayMode),
+                        }}
+                    >
+                        <NetworkModificationTreePane
+                            studyUuid={studyUuid}
+                            studyMapTreeDisplay={studyDisplayMode}
+                        />
+                    </div>
+                    <div
+                        style={{
+                            display:
+                                studyDisplayMode === STUDY_DISPLAY_MODE.TREE
+                                    ? 'none'
+                                    : 'flex',
+                            flexDirection: 'row',
+                            flexGrow: 1,
+                            width: '100%',
+                        }}
+                    >
+                        <Box
+                            sx={styles.map}
+                            style={{
+                                display:
+                                    studyDisplayMode === STUDY_DISPLAY_MODE.TREE
+                                        ? 'none'
+                                        : null,
+                                width: getMapWitdh(studyDisplayMode),
+                            }}
+                        >
+                            <Box>
+                                <Box sx={styles.mapBelowDiagrams}>
+                                    <NetworkMapTab
+                                        networkMapRef={networkMapref}
```

Comment:
move the ref creation inside NetworkMapTab and add a prop to NetworkMapTab to reset the draw mode for example which will call networkMapref.current.getMapDrawer().deleteAll();